### PR TITLE
【hydra No.8】Adapt ldc2d_steady_Re10 to hydra

### PR DESCRIPTION
**Pull request by lknt**
### PR types

New features

### PR changes

Others

### Describe

1、使用第三方库 [hydra](https://hydra.cc/docs/0.11/intro/)适配 ldc2d_steady_Re10 案例
2、ldc2d_steady_Re10 案例代码重构，添加评估函数 evaluate，及其训练/评估命令和预训练模型、指标

### CODE REVIEW

File: docs/zh/examples/ldc2d_steady.md [hunk 1]
```diff
@@ -201,20 +206,16 @@ examples/ldc/ldc2d_steady_Re10.py:39:50
...
+    pde = ppsci.constraint.InteriorConstraint(
+        equation["NavierStokes"].equations,
+        {"continuity": 0, "momentum_x": 0, "momentum_y": 0},
+        geom["rect"],
+        {**train_dataloader_cfg, "batch_size": NPOINT_PDE},
+        ppsci.loss.MSELoss("sum"),
+        evenly=True,
+        weight_dict=cfg.TRAIN.weight.pde, # (1)
+        name="EQ",
+    )
```
Comment (HydrogenSulfate) [inline docs/zh/examples/ldc2d_steady.md:218]: 往左缩进4个空格

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/ldc/ldc2d_steady_Re10.py [hunk 2]
```diff
@@ -50,18 +49,14 @@
...
         geom["rect"],
         {**train_dataloader_cfg, "batch_size": NPOINT_PDE},
         ppsci.loss.MSELoss("sum"),
         evenly=True,
-        weight_dict={
-            "continuity": 0.0001,
-            "momentum_x": 0.0001,
-            "momentum_y": 0.0001,
-        },
+        weight_dict=cfg.TRAIN.weight.pde,  # (1)
```
Comment (HydrogenSulfate) [inline examples/ldc/ldc2d_steady_Re10.py:59]: 末尾的# (1)可以删掉

File: examples/ldc/ldc2d_steady_Re10.py [hunk 3]
```diff
@@ -102,20 +97,17 @@
...
     lr_scheduler = ppsci.optimizer.lr_scheduler.Cosine(
-        EPOCHS,
-        ITERS_PER_EPOCH,
-        0.001,
-        warmup_epoch=int(0.05 * EPOCHS),
+        **cfg.TRAIN.lr_scheduler,
+        warmup_epoch=int(0.05 * cfg.TRAIN.epochs),
     )()
 
     # set optimizer
```
Comment (HydrogenSulfate) [inline examples/ldc/ldc2d_steady_Re10.py:113]: 删除 `# set training hyper-parameters` 注释，并改成这样：
``` py
    # set optimizer
    lr_scheduler = ppsci.optimizer.lr_scheduler.Cosine(
        **cfg.TRAIN.lr_scheduler,
        warmup_epoch=int(0.05 * cfg.TRAIN.epochs),
    )()
    optimizer = ppsci.optimizer.Adam(lr_scheduler)(model)
```

File: examples/ldc/ldc2d_steady_Re10.py [hunk 4]
```diff
@@ -156,13 +148,13 @@
...
-        ITERS_PER_EPOCH,
-        eval_during_train=True,
-        eval_freq=200,
+        cfg.TRAIN.epochs,
+        cfg.TRAIN.iters_per_epoch,
+        eval_during_train=cfg.EVAL.pretrained_model_path,
+        eval_freq=cfg.TRAIN.eval_freq,
         equation=equation,
         geom=geom,
         validator=validator,
```
Comment (HydrogenSulfate) [inline examples/ldc/ldc2d_steady_Re10.py:160]: 末尾加上一行参数：`checkpoint_path=cfg.TRAIN.checkpoint_path,`

File: docs/zh/examples/ldc2d_steady.md [hunk 5]
```diff
@@ -245,47 +246,47 @@ pde_constraint = ppsci.constraint.InteriorConstraint(
...
 --8<--
 ```
 
 ### 3.5 超参数设定
 
-接下来我们需要指定训练轮数和学习率，此处我们按实验经验，使用两万轮训练轮数和带有 warmup 的 Cosine 余弦衰减学习率。
+接下来需要在配置文件中指定训练轮数，此处我们按实验经验，使用两万轮训练轮数和带有 warmup 的 Cosine 余弦衰减学习率。
 
-``` py linenums="112"
+``` py linenums="39"
```
Comment (HydrogenSulfate) [inline docs/zh/examples/ldc2d_steady.md:267]: ==> ``` yaml

File: docs/zh/examples/ldc2d_steady.md [hunk 6]
```diff
@@ -245,47 +246,47 @@ pde_constraint = ppsci.constraint.InteriorConstraint(
...
 
 训练过程会调用优化器来更新模型参数，此处选择较为常用的 `Adam` 优化器。
 
-``` py linenums="121"
+``` py linenums="113"
 --8<--
-examples/ldc/ldc2d_steady_Re10.py:121:122
+examples/ldc/ldc2d_steady_Re10.py:113:114
 --8<--
 ```
```
Comment (HydrogenSulfate) [inline docs/zh/examples/ldc2d_steady.md:281]: 包含整个`# set optimizer`段落

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

**Review by HydrogenSulfate — APPROVED** [review]
LGTM
